### PR TITLE
[READY_FOR_REVIEW]: Update Bebop dex-aggregator

### PR DESCRIPTION
**Pull request by B1boid**
**New chains for existing version**
Just copied the existing code and replaced the blockchain name.
1) Version: "2" (rfq)
Optimism - bebop_rfq_optimism_trades.sql
2) Version: "jam"
Base, Zksync, Scroll, Optimism - bebop_jam_chain_trades.sql

**New version**
Added new version "blend" which is next version of RFQ(version: "2")
For chains: Ethereum, Base, Arbitrum, Polygon

notes about naming: taker=user, maker=market-maker (since protocol is using market-maker liquidity for swaps)

### CODE REVIEW

File: models/bebop/arbitrum/bebop_blend_arbitrum_trades.sql [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        UNION ALL
+        SELECT
+            call_success, call_block_time, call_block_number, call_tx_hash, contract_address, "order",
+            ROW_NUMBER() OVER (PARTITION BY call_tx_hash ORDER BY call_block_number) AS row_num
+        FROM {{ source('bebop_pmms_arbitrum', 'BebopSettlement_call_swapAggregate') }}
+        ) ex
+        ON ex.call_tx_hash = evt.evt_tx_hash and ex.row_num = evt.row_num
+    WHERE ex.call_success = TRUE
+    {% if is_incremental() %}
+    AND evt.evt_block_time >= date_trunc('day', now() - interval '7' Day)
```
Comment (Hosuke) [inline models/bebop/arbitrum/bebop_blend_arbitrum_trades.sql:157]: Same as above.

File: models/bebop/arbitrum/bebop_blend_arbitrum_trades.sql [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+  t.evt_index
+FROM (SELECT * FROM bebop_single_trade UNION ALL SELECT * FROM bebop_multi_and_aggregate_trades) t
+INNER JOIN 
+{{ source('arbitrum', 'transactions')}} tx
+    ON t.tx_hash = tx.hash
+    {% if not is_incremental() %}
+    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc('day', now() - interval '7' Day)
```
Comment (Hosuke) [inline models/bebop/arbitrum/bebop_blend_arbitrum_trades.sql:273]: Same as above.

File: models/bebop/base/bebop_base_trades.sql [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{{ config(
+        schema = 'bebop_base',
+        alias = 'trades',
+        post_hook='{{ expose_spells(\'["base"]\',
+                        "project",
+                        "bebop",
+                        \'["alekss"]\') }}'
```
Comment (Hosuke) [inline models/bebop/base/bebop_base_trades.sql:7]: We can use the new format for post_hook here:
```suggestion
        materialized = 'view',
        post_hook='{{ expose_spells(blockchains = \'["base"]\',
                                    spell_type = "project",
                                    spell_name = "bebop",
                                    contributors = \'["alekss"]\') }}'
```

File: models/bebop/base/bebop_blend_base_trades.sql [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+        UNION ALL
+        SELECT
+            call_success, call_block_time, call_block_number, call_tx_hash, contract_address, "order",
+            ROW_NUMBER() OVER (PARTITION BY call_tx_hash ORDER BY call_block_number) AS row_num
+        FROM {{ source('bebop_pmms_base', 'BebopSettlement_call_swapAggregate') }}
+        ) ex
+        ON ex.call_tx_hash = evt.evt_tx_hash and ex.row_num = evt.row_num
+    WHERE ex.call_success = TRUE
+    {% if is_incremental() %}
+    AND evt.evt_block_time >= date_trunc('day', now() - interval '7' Day)
```
Comment (Hosuke) [inline models/bebop/base/bebop_blend_base_trades.sql:157]: Same as above incremental filters.

**Review by Hosuke — CHANGES_REQUESTED** [review]
We can use `incremental_predicate` for incremental filters.
Also there is a new format for the post_hook.

**Comment by B1boid** [comment]
> We can use `incremental_predicate` for incremental filters. Also there is a new format for the post_hook.

Thanks, fixed

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @B1boid 

[Sample bebop.trades]()

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**